### ISSUE TITLE
Doesn't see stopped containers on restart

### ISSUE THREAD

**Issue by simmerz** [sosedoff/docker-router#20]
Sometimes docker needs to be updated, or the system rebooted with a new kernel.

Because docker containers "managed" by the router need to be run without `--restart=always`, they don't start again when docker is restarted. But as a result, they aren't loaded into `docker-router`'s list of containers to manage. Subsequently, those that have stopped due to a timeout are now dead and never available again unless restarted manually.

**Comment by sosedoff** [comment]
What are the steps to reproduce this locally? Restart the docker daemon?

**Comment by sosedoff** [comment]
Just tried to reproduce the issue - i have 1 router container and 1 app container with idle time of 1 minute. Made a few requests (all goo) then it was killed by router. After that i restarted the docker daemon then opened the same url and got the http response, so the "managed" containers are in fact get started. Am i missing some steps here?

**Comment by simmerz** [comment]
I'll go back and figure out what I'm missing out. Clearly doing something wrong/different.

**Comment by simmerz** [comment]
So I have my router and a container that was stopped. When I restart the docker service, the managed container gets started, but the router responds with a 503. If I then restart the docker router, all works fine.

**Comment by sosedoff** [comment]
Can you provide some logs?

**Comment by simmerz** [comment]
Typical.. now I try and do it to fetch logs and it works perfectly. I'll close with this comment and reopen if it happens again and I can get some logging data.

**Comment by sosedoff** [comment]
I think i might have an idea: when the daemon restarts (or when the idle container restarts) the container is brought back online but the router does not serve the initial request (for some reason), which results in 503 error. any subsequent requests are served correctly though. maybe this is the behavior you're seeing

**Comment by simmerz** [comment]
No, it was consistently returning a 503. Now I can't replicate it again, so perhaps it was a config issue at my end, though I have no idea what!

**Comment by simmerz** [comment]
@sosedoff I think I've finally figured out what happens here: A container is started, but it is never accessed before it gets switched off by the router. Subsequent requests never seem to restart it.

**Comment by simmerz** [comment]
@sosedoff this one is still causing me a bit of a headache and not sure where to look to resolve it. I'm guessing you haven't been able to replicate it.

Could you share an example command you use to start a container with a timeout that you don't touch for a while that then just works when you come back to it some while later?

I've tried adding in a healthcheck path but that hasn't made any difference. If I hit the URL within 30 minutes it works (the timeout) and then it fails beyond that.

If I hit the URL in the 30 minute window though, it'll fire up properly after it's been stopped by the router.

**Comment by sosedoff** [comment]
Oh i was def able to replicate your issue, i just didnt have any time to fix it since im not using the feature myself so there's no real urgency :)

**Comment by simmerz** [comment]
Ok good to know it's not just me then.
